lib/buffer.js

```diff
@@ -561,14 +557,6 @@ Buffer.prototype.write = function(string, offset, length, encoding) {
       encoding = length;
       length = undefined;
     }
-
-  // XXX legacy write(string, encoding, offset, length) - remove in v0.13
-  } else {
```

Comment:
`Buffer.fill(val, start, end)` just coerces `start` and `end` to integers. This does the same with `offset`. Any calls that still use the old arguments `(string, encoding, offset, length)` will fail with an invalid encoding exception.

I can make it throw `new Error("Buffer.write(string, encoding, offset, length) is no longer supported")` -- do you prefer that?